File: server/src/main/java/com/lordbyronsenterprises/server/user/AddressService.java
package com.lordbyronsenterprises.server.user;

import java.util.List;

public interface AddressService {
    List<AddressDto> getAddressesForUser(User user);
    AddressDto createAddress(User user, AddressDto addressDto);
    AddressDto updateAddress(User user, Long addressId, AddressDto addressDto);
    void deleteAddress(User user, Long addressId);
}

File: client/src/api/addressService.js
import api from "./axiosConfig";

export const getMyAddresses = async () => {
  const response = await api.get("/user/addresses");
  return response.data;
};

export const createAddress = async (addressData) => {
  const response = await api.post("/user/addresses", addressData);
  return response.data;
};

export const updateAddress = async (addressId, addressData) => {
  const response = await api.put(`/user/addresses/${addressId}`, addressData);
  return response.data;
};

export const deleteAddress = async (addressId) => {
  await api.delete(`/user/addresses/${addressId}`);
};
